WordPress/Classes/ViewRelated/Menus/MenusViewController.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+extension MenusViewController {
+
+    /// Fetch all pages from the site.
+    ///
+    /// - Returns: A block that can be used to cancel the fetching.
+    @objc func fetchAllPages(success: @escaping ([MenuItem]) -> Void, failure: @escaping (Error) -> Void) -> () -> Void {
+        let coreDataStack = ContextManager.shared
+        let repository = PostRepository(coreDataStack: coreDataStack)
+        let fetchAllPagesTask = repository.fetchAllPages(statuses: [.publish], in: TaggedManagedObjectID(blog))
+
+        // Wait for the fetching all pages task to complete and pass its result to the success or failure block.
+        Task { [weak self] in
+            let allPages: [TaggedManagedObjectID<Page>]
+            do {
+                allPages = try await fetchAllPagesTask.value
+            } catch {
+                failure(error)
+                return
+            }
+
+            guard let menusService = self?.menusService else { return }
```

Comment:
I was going to suggest moving the check at the start of the `Task` to exit early. But maybe it's best to check if `self` is still available after the `fetchAllPagesTask` completed, in case the ViewController (`self`) was deallocated in the meantime. Is this the reason for the check being at this point?